From e388c9829ac7f723d05ac8d8897b0e72d5d93c56 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Wed, 1 Mar 2017 17:37:10 -0800
Subject: [PATCH 2317/2848] HADOOP-14131. kms.sh creates bogus dir for tomcat
 logs. Contributed by John Zhuge.

Change-Id: I444e10318eb64dcd4c0416b985d12049c72e8943
(cherry picked from commit 133857ba049e4b74579952141b6b03cad37df2a0)
---
 .../hadoop-kms/src/main/sbin/kms.sh                |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh b/hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh
index a7878fc..c3edf0c 100644
--- a/hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh
+++ b/hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh
@@ -56,7 +56,8 @@ fi
 print "Using   CATALINA_OPTS:       ${CATALINA_OPTS}"
 
 catalina_opts="-Dproc_kms"
-catalina_opts="${catalina_opts} -Djava.library.path=${JAVA_LIBRARY_PATH}";
+catalina_opts="${catalina_opts} -Dkms.log.dir=${KMS_LOG}"
+catalina_opts="${catalina_opts} -Djava.library.path=${JAVA_LIBRARY_PATH}"
 
 print "Adding to CATALINA_OPTS:     ${catalina_opts}"
 print "Found KMS_SSL_KEYSTORE_PASS:     `echo ${KMS_SSL_KEYSTORE_PASS} | sed 's/./*/g'`"
@@ -81,7 +82,6 @@ if [[ "${1}" = "start" || "${1}" = "run" ]]; then
   catalina_init_properties
   catalina_set_property "kms.home.dir" "${KMS_HOME}"
   catalina_set_property "kms.config.dir" "${KMS_CONFIG}"
-  catalina_set_property "kms.log.dir" "${KMS_LOG}"
   catalina_set_property "kms.temp.dir" "${KMS_TEMP}"
   catalina_set_property "kms.admin.port" "${KMS_ADMIN_PORT}"
   catalina_set_property "kms.http.port" "${KMS_HTTP_PORT}"
-- 
1.7.9.5

